src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Unity/Extension/UnsafeBlitFormatter.cs

```diff
@@ -33,8 +33,9 @@ public void Serialize(ref MessagePackWriter writer, T[] value, MessagePackSerial
             }
 
             var byteLen = value.Length * Marshal.SizeOf<T>();
+            var realLen = MessagePackWriter.IntegerEncodedLength(byteLen) + byteLen + 1;
```

Comment:
As the issue you opened calls out, this is one of two possible fixes. You're taking the safer fix here, which may be the right one. But as it is more complex than removing `byteLen` from the extension's payload altogether given it's redundant, I'd like to entertain the alternative. @neuecc what do you think? Is this formatter used for persisting data or just volatile memory communication such that a breaking change in the format wouldn't be too impactful?

Reply:
In my use case it's for persisting data, so I imagine someone else is doing the same. I'd like to hear neuecc's ideas too.

Reply:
In lieu of any response from @neuecc, let's take the safe route.